src/lib/jsonParseV2/parseContextAttributeCompoundValue.cpp

```diff
@@ -325,3 +325,144 @@ std::string parseContextAttributeCompoundValue
+  orion::ValueType     valueType
+)
+{
+  caP->compoundValueP            = new orion::CompoundValueNode();
+  caP->compoundValueP->name      = "TOP";
+  caP->compoundValueP->container = caP->compoundValueP;
+  caP->compoundValueP->valueType = caP->valueType;  // Convert to other type?
+  caP->compoundValueP->path      = "/";
+  caP->compoundValueP->rootP     = caP->compoundValueP;
+  caP->compoundValueP->level     = 0;
+  caP->compoundValueP->siblingNo = 0;
+
+  orion::CompoundValueNode*   parent  = caP->compoundValueP;
+
+
+  //
+  // Children of the node
+  //
+  if (caP->valueType == orion::ValueTypeVector)
+  {
+    int counter  = 0;
+
+    for (Value::ConstValueIterator iter = document.Begin(); iter != document.End(); ++iter)
+    {
+      std::string                nodeType  = jsonParseTypeNames[iter->GetType()];
+      orion::CompoundValueNode*  cvnP      = new orion::CompoundValueNode();
+      char                       itemNo[4];
+
+      snprintf(itemNo, sizeof(itemNo), "%03d", counter);
+
+      cvnP->valueType  = stringToCompoundType(nodeType);
+      cvnP->container  = parent;
+      cvnP->rootP      = parent->rootP;
+      cvnP->level      = parent->level + 1;
+      cvnP->siblingNo  = counter;
+      cvnP->path       = parent->path + "[" + itemNo + "]";
+
+
+      if (nodeType == "String")
+      {
+        cvnP->stringValue       = iter->GetString();
+      }
+      else if (nodeType == "Number")
+      {
+        cvnP->numberValue = iter->GetDouble();
+      }
+      else if ((nodeType == "True") || (nodeType == "False"))
+      {
+        cvnP->boolValue   = (nodeType == "True")? true : false;
+      }
+      else if (nodeType == "Object")
+      {
+        cvnP->path += "/";
+        cvnP->valueType = orion::ValueTypeObject;
+      }
+      else if (nodeType == "Array")
+      {
+        cvnP->path += "/";
+        cvnP->valueType = orion::ValueTypeVector;
+      }
+
+      parent->childV.push_back(cvnP);
+
+      //
+      // Recursive call if Object or Array
+      //
+      if ((nodeType == "Object") || (nodeType == "Array"))
+      {
+        parseContextAttributeCompoundValue(iter, caP, cvnP);
+      }
+
+      ++counter;
+    }
+  }
+  else if (caP->valueType == orion::ValueTypeObject)
+  {
+    int counter  = 0;
+
+    for (Value::ConstMemberIterator iter = document.MemberBegin(); iter != document.MemberEnd(); ++iter)
+    {
+      std::string                nodeType = jsonParseTypeNames[iter->value.GetType()];
+      orion::CompoundValueNode*  cvnP     = new orion::CompoundValueNode();
+
+      cvnP->name       = iter->name.GetString();
+      cvnP->valueType  = stringToCompoundType(nodeType);
+      cvnP->container  = parent;
+      cvnP->rootP      = parent->rootP;
+      cvnP->level      = parent->level + 1;
+      cvnP->siblingNo  = counter;
+      cvnP->path       = parent->path + cvnP->name;
+
+      if (nodeType == "String")
+      {
+        cvnP->stringValue       = iter->value.GetString();
```

Comment:
Oh yes ...
All 'cvnP->xxxValue =' were supposed to be horizontally aligned. Don't know what happened to this one ...